Source/WebCore/dom/ViewTransition.h

```diff
@@ -52,7 +52,9 @@ enum class ViewTransitionPhase : uint8_t {
 struct CapturedElement {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    RefPtr<ImageBuffer> oldImage;
+    // std::nullopt represents an non-capturable element.
+    // nullptr represents an absent snapshot on an capturable element.
+    std::optional<RefPtr<ImageBuffer>> oldImage;
```

Comment:
This kind of 3 states value would ideally have a proper variant

```
struct NonCapturable {}
struct AbsentSnapshot {}
std::variant<NonCapturable, AbsentSnapshot, Ref<ImageBuffer>>;
```

(not for this PR though)

Reply:
I don't think the variant is a great idea, `std::optional` reads more like the spec, because lines like if "old image is not null" read exactly as `if (capturedElement->oldImage)`.